File: challenge_130_easy_roll_the_dies/C/roll_the_dice.c
/**********************************************************/
/* This file is published under the 3-clause BSD license. */
/* Copyright 2014 Manolis Fragkiskos Ragkousis		  */
/* Check the license file.				  */
/**********************************************************/
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <time.h>

int 
main(int argc, char **argv)
{
  const char s[2]="d";
  char *token;
  int num_of_dices;
  int num_of_faces;
  
  if(!(token = strtok(argv[1], s)))
    {
      puts("wrong input, insert NdM");
      exit(-1);
    }
  num_of_dices = atoi(token);
  
  token = strtok(NULL, s);
  num_of_faces = atoi(token);
  
  printf("%dd%d dice(s) roll... ",num_of_dices,num_of_faces);

  srand(time(NULL));
  for(int i=0;i<num_of_dices;i++)
    printf("%d ",(rand() % num_of_faces)+1);
  printf("\n");

  return 0;
}

File: challenge_130_easy_roll_the_dies/python/roll_the_dice.py
#!/usr/bin/python3.4
# This file is published under the 3-clause BSD license
# Copyright 2014 Manolis Fragkiskos Ragkousis
# Check the license file.	
import sys
import random

dice = sys.argv[1].split("d");

num_of_dices = int(dice[0]);
num_of_faces = int(dice[1]);

if(num_of_faces <=0 or num_of_faces >100):
    print("Dice faces must be between 1-100"),
    exit(-1);

print("%dd%d dice(s) roll... " %(num_of_dices,num_of_faces));

count=0;
while(count < num_of_dices):
    print(random.randint(1, num_of_faces),end=' '),
    count = count + 1
print("");
    
